src/text-editor.js

```diff
@@ -3881,6 +3903,11 @@ class TextEditor {
     return this.getLastCursor().getScopeDescriptor()
   }
 
+  // Get the syntax nodes at the cursor.
+  getSyntaxTree () {
```

Comment:
This method name should probably start with `getCursor`, to make it clear that the positioning is derived from the cursor. Maybe `getCursorSyntaxTreePath`.